common/predictive-text/worker/index.ts

```diff
@@ -48,9 +55,50 @@ class LMLayerWorker {
    * 
    *   // Don't do this!
    *   self.onmessage = worker.onMessage;
+   * 
+   * See: .install();
+   */
+  onMessage(event: MessageEvent) {
+    const {message} = event.data;
+    if (!message) {
+      throw new Error(`Missing required 'message' attribute: ${event.data}`)
+    }
+
+    // Load the model.
+    let model = this._importScripts(event.data.model);
+
+    this.cast('ready', {
+      configuration: {
+        // Send a reasonable, but non-configurable amount for now.
+        leftContextCodeUnits: 64,
+        rightContextCodeUnits: 0,
+      }
+    });
+  }
+
+  /**
+   * Sends back a message structured according to the protocol.
+   * @param message A message type.
+   * @param payload The message's payload. Can have any properties, except 'message'.
    */
-  onMessage() {
-    this._postMessage({ message: 'ready' });
+  private cast(message: OutgoingMessage, payload: object) {
+    this._postMessage({ message, ...payload });
+  }
+
+  /**
+   * Creates a new instance of the LMLayerWorker, and installs
+   * all its functions within the provided Worker scope.
```

Comment:
This could use somewhat better documentation for clarity; its real purpose is to create and install _the_ LMLayerWorker instance as the core of a Worker thread, connecting the global Worker scope with that of the class where necessary.  (I saw the phrase "a new instance" and incorrectly inferred that we'd want multiple of these running around in the final, production version.)

That said, the method definitely makes sense, and it's clearly helpful for rigging the Worker core up for mocked testing.  Excellent!